resteasy-client-api/src/main/java/org/jboss/resteasy/client/exception/WebApplicationExceptionWrapper.java

```diff
@@ -0,0 +1,151 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ *
+ * Copyright 2020 Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jboss.resteasy.client.exception;
+
+import javax.ws.rs.BadRequestException;
+import javax.ws.rs.ClientErrorException;
+import javax.ws.rs.ForbiddenException;
+import javax.ws.rs.InternalServerErrorException;
+import javax.ws.rs.NotAcceptableException;
+import javax.ws.rs.NotAllowedException;
+import javax.ws.rs.NotAuthorizedException;
+import javax.ws.rs.NotFoundException;
+import javax.ws.rs.NotSupportedException;
+import javax.ws.rs.RedirectionException;
+import javax.ws.rs.ServerErrorException;
+import javax.ws.rs.ServiceUnavailableException;
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+import org.eclipse.microprofile.config.Config;
+import org.eclipse.microprofile.config.ConfigProvider;
+import org.jboss.resteasy.spi.ResteasyDeployment;
+
+/**
+ * An interface which allows a {@link WebApplicationException} to be unwrapped.
+ *
+ * @author <a href="mailto:jperkins@redhat.com">James R. Perkins</a>
+ */
+public interface WebApplicationExceptionWrapper<T extends WebApplicationException> {
+
+    /**
+     * If the {@code resteasy.original.webapplicationexception.behavior} is set to {@code true} or the request is
+     * determined to not be a server side request, then the {@link WebApplicationException} passed in will be returned.
+     * If the property is not set to {@code true} and this is a server side request then the exception is wrapped and
+     * the response is {@linkplain #sanitize(Response) sanitized}.
+     *
+     * @param e the exception to possibly wrapped
+     *
+     * @return the wrapped exception or the original exception if the exception has already been wrapped the the
+     * wrapping feature is turned off
+     */
+    static WebApplicationException wrap(final WebApplicationException e) {
+        final Config config = ConfigProvider.getConfig();
+        final boolean originalBehavior = config.getOptionalValue("resteasy.original.webapplicationexception.behavior", boolean.class).orElse(false);
```

Comment:
Thinking about this, would this take into account context parameters from a `web.xml`?